diff --git a/dist/index.mjs b/dist/index.mjs
index 01188992e58fb403216ab3ad43d56db66401f65d..1b271fe84f5a11fe574bb93047b536be077770ec 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -551,7 +551,11 @@ function emitLegacyIndexDeprecation(url, packageJsonUrl, base, main) {
   if (process$1.noDeprecation) {
     return;
   }
-  const format = defaultGetFormatWithoutErrors(url, { parentURL: base.href });
+  const baseHref = base && (typeof base === 'object' ? base.href : base);
+  if (!baseHref) {
+    return;
+  }
+  const format = defaultGetFormatWithoutErrors(url, { parentURL: baseHref });
   if (format !== "module") return;
   const urlPath = fileURLToPath(url.href);
   const packagePath = fileURLToPath(new URL$1(".", packageJsonUrl));
